paper/paper.md

```diff
@@ -24,11 +24,16 @@ bibliography: paper.bib
-However, existing neural networks that use 32 bits to encode each weight and activation have an energy budget which is far beyond the scope of many of these applications. One common way to improve computational efficiency is to reduce the precision of the network to 16-bit or 8-bit, also known as quantization.
-Binarized Neural Networks (BNNs) represent an extreme case of quantized networks, that cannot be viewed as approximations to real-valued networks and therefore requires special tools and optimization strategies [@bop]. In these networks both weights and activations are restricted to $\{-1, +1\}$ [@binarynet]. Compared to an equivalent 8-bit quantized network BNNs require 8 times less memory and can lead to a theoretical 64 times speed-up when deployed on optimized hardware.
+Modern deep learning methods have been successfully applied to many different tasks and have the potential to revolutionize everyday lives. However, existing neural networks that use 32 bits to encode each weight and activation often have an energy budget that far exceeds the capabilities of mobile or embedded devices. One common way to improve computational efficiency is to reduce the precision of the network to 16-bit or 8-bit, also known as quantization.
+Binarized Neural Networks (BNNs) represent an extreme case of quantized networks, that cannot be viewed as approximations to real-valued networks and therefore requires special tools and optimization strategies [@bop]. In these networks both weights and activations are restricted to $\{-1, +1\}$ [@binarynet]. Compared to an equivalent 8-bit quantized network BNNs require 8 times smaller memory size and 8 times fewer memory accesses, which reduces energy consumption drastically when deployed on optimized hardware [@binarynet].
 However, many open research questions remain until the use of BNNs and other extremely quantized neural networks becomes widespread in industry. [`larq`](https://larq.dev) is an ecosystem of Python packages for BNNs and other Quantized Neural Networks (QNNs). It is intended to facilitate researchers to resolve these outstanding questions.
 
+# Statement of need
+
+Fortunately, many researchers already publish code along with their papers, though, in absence of a common API to define extremely quantized networks, authors end up re-implementing a large amount of code, making it difficult to share improvements and make rapid progress. Existing tools for BNN research either focus on deployment or require a high learning curve to use and contribute to [@dabnn; @bmxnetv2]. The API of [`larq`](https://larq.dev) is built on top of `tensorflow.keras` [@tensorflow; @keras] and is designed to provide an easy to use, composable way to design and train BNNs. While popular libraries like TensorFlow Lite, TensorFlow Model Optimization or PyTorch focus on 16-bit or 8-bit quantization [@tensorflow; @pytorch], `larq` aims to extend this towards lower bit-widths.
```

Comment:
Perhaps we should link to some example repos here?

Reply:
> Perhaps we should link to some example repos here?

I am not sure about that, what examples do you have in mind?

Reply:
The BinaryNet repos [1](https://github.com/itayhubara/BinaryNet), [2](https://github.com/itayhubara/BinaryNet.tf), [3](https://github.com/itayhubara/BinaryNet.pytorch), [4](https://github.com/MatthieuCourbariaux/BinaryNet) and [birealnet](https://github.com/liuzechun/Bi-Real-net) cover a pretty wide range of ML frameworks. More recent options [1](https://github.com/bczhangbczhang/Bayesian-Optimized-1-bit-CNNs) [2](https://github.com/mi-lad/studying-binary-neural-networks) [3](https://github.com/AcrossV/Gated-XNOR) [4](https://github.com/bohanzhuang/Group-Net-object-detection)

Reply:
Do you think it would add much to the paper? I'd be happy to include the links as references here.

Reply:
not sure, I would prefer to see references after the first sentence but without a "5-14" type referencing style it may take more space than it's worth.

Reply:
> but without a "5-14" type referencing style it may take more space than it's worth.

Yes, I think it would take up too much space for what it is worth :+1: